akka-remote/src/main/scala/akka/remote/Remoting.scala

```diff
@@ -372,8 +372,13 @@ private[remote] object EndpointManager {
      */
     def markAsFailed(endpoint: ActorRef, timeOfRelease: Deadline): Unit =
       if (isWritable(endpoint)) {
-        addressToWritable += writableToAddress(endpoint) -> Gated(timeOfRelease)
-        writableToAddress -= endpoint
+        val address = writableToAddress(endpoint)
+        addressToWritable.get(address) match {
+          case Some(Quarantined(_, _)) ⇒ // don't overwrite Quarantined with Gated
+          case _ ⇒
```

Comment:
Quarantine piercing lose quarantine marker (e.g. the state is ```Pass(_,_, Some(uid)```) ? and seems redundance when used with ```keepQuarantinedOr``` method together in the EM's ```supervisorStrategy```?